Copy go/v1beta1 files to go/v1

noahklein opened the pull request.

aysylu commented on go/v1/api/grafeas.go:
```diff
@@ -0,0 +1,156 @@
+import (
+	"github.com/grafeas/grafeas/go/errors"
+	"github.com/grafeas/grafeas/go/iam"
+	gpb "github.com/grafeas/grafeas/proto/v1beta1/grafeas_go_proto"
```
This should be v1, right?

aysylu commented on go/v1/api/grafeas.go:
```diff
@@ -0,0 +1,156 @@
+	Errorf(ctx context.Context, format string, args ...interface{})
+}
+
+// API implements the methods in the v1beta1 Grafeas API.
```
s/v1beta1/v1 everywhere appropriate please

aysylu commented on go/v1/api/validators/vulnerability/vulnerability_test.go:
```diff
@@ -0,0 +1,329 @@
+	"testing"
+
+	pkgpb "github.com/grafeas/grafeas/proto/v1beta1/package_go_proto"
+	vulnpb "github.com/grafeas/grafeas/proto/v1beta1/vulnerability_go_proto"
```
I think you'll need to generate these as well.

noahklein commented:
* Updated Makefile to build the v1 go proto libraries.
* Built the v1 go proto libraries.
* Updated V1 imports to point to v1 dependencies.

@aysylu, @vtsao PTAL

aysylu commented on Makefile:
```diff
@@ -45,57 +47,20 @@ grafeas_go_v1alpha1: .install.protoc-gen-go .install.grpc-gateway v1alpha1/proto
-define gen_go_proto
-	protoc/bin/protoc \
-		-I ./ \
+%_go_proto: %.proto protoc/bin/protoc install.tools
```
Maybe add a high-level comment above for what this does?

aysylu commented on go/v1/api/validators/README.md:
```diff
@@ -0,0 +1,6 @@
+
+When making changes to these validators, please ensure to update the
+documentation on the protos so they match the code. E.g., if what fields are
+required change for a note resource, please update the note message proto's
```
I'm not sure how to parse this part, could you please rephrase?

aysylu commented on proto/v1/swagger/image.swagger.json:
```diff
@@ -0,0 +1,19 @@
+{
```
Let's remove all the swagger specs for now, as we might not be generating client libraries from Swagger spec going forward.

aysylu commented on proto/v1beta1/swagger/grafeas.swagger.json:
```diff
@@ -15,13 +15,13 @@
   ],
   "paths": {
-    "/v1beta1/{name}": {
+    "/v1beta1/{name=projects/*/notes/*}": {
```
Was this auto-generated or manually added? I don't believe this path would actually be interpreted correctly by swagger because OpenAPI doesn't support paths with wildcards in them.

vtsao replied:
These are likely caused by a new version of protoc that we've pulled in because I don't think we lock in a specific version of protoc to use. So these changes should be fine.

aysylu commented on proto/v1beta1/swagger/grafeas.swagger.json:
```diff
@@ -15,13 +15,13 @@
         "responses": {
           "200": {
-            "description": "",
+            "description": "A successful response.",
```
Revert this change and the ones below? This should be unchanged, right?

aysylu commented on proto/v1beta1/swagger/project.swagger.json:
```diff
@@ -21,7 +21,7 @@
         "responses": {
           "200": {
-            "description": "",
+            "description": "A successful response.",
```
This file should be unchanged by the PR, right? If so, let's revert all the previous changes.

aysylu commented on v1alpha1/proto/grafeas.swagger.json:
```diff
@@ -17,11 +17,11 @@
     "/v1alpha1/projects": {
       "get": {
-        "summary": "Deletes the given `Occurrence` from the system. Use this when\nan `Occurrence` is no longer applicable for the given resource.",
+        "summary": "Lists projects.",
```
Does this PR affect this file? If not, let's revert the changes here.
